lib/src/rules/unnecessary_parenthesis.dart

```diff
@@ -100,11 +101,22 @@ class _Visitor extends SimpleAstVisitor<void> {
       }
     }
 
-    if (parent is ParenthesizedExpression) {
+    // Directly wrapped into parentheses already - always report.
+    if (parent is ParenthesizedExpression ||
+        parent is InterpolationExpression ||
+        (parent is ArgumentList && parent.arguments.length == 1) ||
```

Comment:
It isn't clear to me why this is only true when there's a single argument. Are there valid reasons for allowing `o.m(a, (b), c);`?

Also, do we want to handle the case of named arguments (`0.m(a: (b));`)?

Reply:
Yes, `o.m(a, (b), c);` are currently not handled and will still not be handled.

**But** this is intentionally written like this because this whole set of conditions aims to detect **specifically** `((b))` and nothing else, and that's also the only rule I'm arguing for within this proposal.

Reply:
Keeping this scoped to single args sounds reasonable to me. (Though I do think we should consider cases like `o.m(a, (b), c)` in the future.)